src/helpers/message.ts

```diff
@@ -27,105 +27,161 @@ function _getTokenFmt(amount: BigNumber, token: TokenDto) {
+  } else {
+    return ''
   }
+}
 
-  // Does it expire?
-  const isNeverExpires = isNeverExpiresOrder(validUntil)
-
-  let expirationMsg: string
-  if (isNeverExpires) {
-    expirationMsg = '\n  - *Expires*: Valid until cancelled'
+export function buildSellMsg(
+  isUnlimited: boolean,
+  buyTokenLabel: string,
+  sellTokenLabel: string,
+  buyAmount: string,
+  sellAmount: string,
+): string {
+  if (isUnlimited) {
+    // doesn't make sense to display amounts when the order is unlimited
+    return `Sell \`${sellTokenLabel}\` for \`${buyTokenLabel}\`\n`
   } else {
-    expirationMsg = `\n  - *Expires*: \`${moment(validUntil).calendar()} GMT\`, \`${moment(validUntil).fromNow()}\``
+    return `Sell *${sellAmount}* \`${sellTokenLabel}\` for *${buyAmount}* \`${buyTokenLabel}\`\n`
   }
-  message += expirationMsg
+}
 
-  if (!sellToken.known || !buyToken.known) {
-    message +=
-      '\n  - "Maybe" means one or more tokens claim to be called as shown, ' +
-      "but it's not currently part of the list of [known tokens](https://github.com/gnosis/dex-js/blob/master/src/tokenList.json). " +
-      'Make sure you verify the address yourself before trading against it.'
-  }
+const ONE = new BigNumber(1)
+
+export function calculateUnlimitedBuyTokenFillAmount(price: BigNumber, sellToken: TokenDto): string {
+  // 1/fee denominator == fee %
+  // fee % * 2 == minimum amount to have a match (both orders paying the fee)
+  const feeToDeduce = ONE.dividedBy(FEE_DENOMINATOR).multipliedBy(2)
+
+  // Since we are inverting the sell/buy to create the 'counter' offer,
+  // we need to invert the price, thus 1/price
+  return (
+    ONE.dividedBy(price)
+      // deduct from the value the fee by multiplying by 1 - FEE
+      .multipliedBy(ONE.minus(feeToDeduce))
+      // scale up the order to have a baseline of selling 10 units of sellToken
+      .multipliedBy(10)
+      // Base 10 to force BigNumber NOT returning value in exponential notation
+      .toString(10)
+      // `rounding` down if needed because the token might not have enough precision.
+      // E.g.: GUSD has only 2 decimal places, thus it makes no sense to send an amount of 9.009
+      .replace(new RegExp(`\\.(\\d{1,${sellToken.decimals}})\\d*`), '.$1')
+  )
+}
+
+export function buildFillOrderMsg(
+  isUnlimited: boolean,
+  order: OrderDto,
+  price: BigNumber,
+  baseUrl: string,
+  buyTokenParam: string,
+  sellTokenParam: string,
+): string {
+  const { buyToken, priceDenominator, sellToken, priceNumerator } = order
 
   let fillAmountBuy: string
   let fillAmountSell: string
   if (isUnlimited) {
-    fillAmountBuy = '10'
-    fillAmountSell = price.multipliedBy(10).multipliedBy(FACTOR_TO_FILL_ORDER)
+    // This is tricky. how much should we offer to fill for a unlimited order?
+    // Going for 10 units
+    fillAmountBuy = calculateUnlimitedBuyTokenFillAmount(price, sellToken)
+    fillAmountSell = '10'
```

Comment:
I think this is rather arbitrary number.

One thinking about this is, the order is unlimited, so we shouldn't use this super-high value that is recorded in the order. It might be reasonable to think that the "balance of the sell token" for that user is a good value for this `fillAmountBuy`, but it can be probably 0 since the user normally deposits after creating the "liquidity orders". 

For an unlimited order, I would say that what we know is the price. There's a suggestion by @c3rnst to add in the trade widget a 3rd input with the price. My suggestion is, leave the amount blank for now (or leave that 10 if you prefer) and include in the params of the URL the price. For now the UI won't use that param, but in the future we can pre-fill this 3rd input so we allow the user to enter just the sellAmount or buyAmount

Reply:
Yep, have exactly the same idea.
I'll leave it at 10 for now (since I already done it) and we can revisit once we decide on the price input.

Reply:
cool